Hide log subscribers & add thirdweb, ethers dependencies

GorillasHabitat opened the pull request:
This commit comments out the lines of code that subscribe to the curator emitter object and log its data for three different topics. The change is to clean up the codebase as the logging is no longer necessary. Also, three new dependencies have been added to the package.json file: '@thirdweb-dev/chains', '@thirdweb-dev/sdk' and 'ethers'. These added dependencies will allow the application to interact with blockchain networks (namely Ethereum), which is necessary for future updates.

GorillasHabitat committed:
"Hide log subscribers & add thirdweb, ethers dependencies"

This commit comments out the lines of code that subscribe to the curator emitter object and log its data for three different topics. The change
is to clean up the codebase as the logging is no longer necessary. Also, three new dependencies have been added to the package.json file: '@thirdweb-dev/chains', '@thirdweb-dev/sdk' and 'ethers'. These added dependencies will allow the application to interact with blockchain networks (namely Ethereum), which is necessary for future updates.

GorillasHabitat committed:
Uncomment subscription code in index.ts

This commit re-enables curator's subscription to topics 'mill.coingecko', 'mill.coinbase' and 'workshop.lsd' in src/index.ts. These subscriptions were previously commented out, but are now required again for the functionality of the application. They will allow curator to process and log data from these topics.

LionsHabitat commented on src/museums.ts:
```diff
@@ -1,48 +1,91 @@
   {
     id: 'tenet-testnet', // required
-    abi: [], // this can default to origami-solidity if not passed
-    seedPhrase: 'hello world', // required unless privateKey
```
You'll still need the seedPhrase and or privateKey, these can change per museum and per collection.  In theory every collection could have their own owner account. We can name it to seedPhraseEnvKey or something like that and pull it from the env on demand.

LionsHabitat commented on src/museums.ts:
```diff
@@ -1,48 +1,91 @@
+    abi: fetch(
+      'https://raw.githubusercontent.com/open-oracle-origami/origami-solidity/main/abi/CollectionImplV1.json'
+    ).then(response => response.json()),
     deviation: 0,
```
As a side note, these could also be configured individually in the collection, the museum top level is the default.

GorillasHabitat committed:
Update dependencies and refactor code

This commit updates all outdated dependencies and refactors some aspects of the `museums.ts` file for clarity and maintainability.

Changed:
- Updated the `origami-js-sdk` dependency to version `^1.0.0-alpha.19`
- Updated the `nodejs` version in `.tool-versions` to `18.17.0`
- Refactored the `museums` configuration in `museums.ts` file. This includes updating hardcoded addresses and removing the extraneous console log.
- Also made changes to how we pass the configuration to `curator.plan`.
- Updated the package-lock.json to correspond to changes in package.json.

This updates were necessary to keep the project dependencies up to date and to improve the maintainability of the code by increasing clarity and removing unnecessary logging.

GorillasHabitat committed:
Update Origami SDK package version

Updated the version of the @open-oracle-origami/origami-js-sdk package from 1.0.0 to 1.0.0-alpha.19. This was a necessary upgrade to ensure compatibility with the latest version of Node (18.17.0).

GorillasHabitat committed:
Refactored curator subscription and expanded museum library

In the src/index.ts file, all curator emitter subscriptions are commented out for clean-up and further use in the future. Simultaneously, in the src/museums.ts file, several enhancements were introduced, including adding types for the SDK contract, expanding CurateConfig, and introducing a function to fetch contract details from the SDK. These changes provide more flexibility and robustness in defining and utilizing museum configurations, leading to more manageable and optimized code.

GorillasHabitat committed:
"Refactor museums.ts for better error handling and code clarity"

This commit refactors several functions in the museums.ts file to improve error handling and overall code clarity. This involves splitting out configurations for museum and museumCollection into two separate functions: getMuseumConfig() and getMuseumCollectionConfig(), to more granularly handle improperly configured collections. fetchLastCertifiedOrigami() is now capable of accepting a collection contract. This helps in fetching the contract’s last update more accurately. certify() method is more generalized and accepts resourceId, which reduces duplicacy and raises readability.

GorillasHabitat committed:
Change fetch method and timestamp handling for origami certification

Modified `fetchLastCertifiedOrigami` to directly call 'latestRoundData' method from the contract, removing unnecessary enforcement and ToDo comment. Also, the timestamp handling in the certification logic has been updated. Now, we're not just checking if there is a previous origami but also checking the timestamp of it. The certification validation has been changed to certify if the last origami is more than 1 hour old instead of the next origami being 1 hour newer. Logging information added to show the remaining hours for origami to be qualified for certification.

GorillasHabitat committed:
Add balance check before creating origami

This commit introduces a check to ensure that there's sufficient balance in the wallet before an origami is created. The transaction hash tracking has also been adjusted for better accuracy, now referencing the transaction receipt instead of transaction object.

GorillasHabitat committed:
Add processing state to museum collections

This update adds a processing state to museum collections for better control over blockchain interactions. A `processing` key was included in museum collection configuration objects, with the default value set as `false`. A set of helper functions were added to check and modify the `processing` state. The `certify` function and `curate` function now implement logic to prevent further actions while the `processing` state is `true` which helps in avoiding race condition issues during asynchronous operations. Culled repeated config id checks and extraneous console log statements for cleaner code.

GorillasHabitat committed:
Refactor museumCollectionConfig and update prettier settings

This commit combines line-by-line operations into single line functions for better code readability in the museums.ts file. Moreover, a configuration change has been made in the prettier settings, where the printWidth parameter has been adjusted from 80 to 120. This change has been made in order to conform more closely with the preferred coding style conventions of the project.

GorillasHabitat committed:
interface Collection {
  id: string
  [key: string]: any
}

interface MuseumConfig {
  id?: string
  abi?: Promise<any>
  deviation?: number
  threshold?: number
  collections: Collection[]
}

GorillasHabitat committed:
Update ESLint configuration and modify museum.ts file

Updated the project's ESLint configuration by removing 'plugin:prettier/recommended' and adding a rule for maximum length of a line. Prettier rules conflict with ESLint's rules often, therefore it's removed and added max-len rule to enforce code readability by having a maximum line length limit.
In the museums.ts file, a newline was added to separate two interface blocks to improve code readability.
